### DISCUSSION TITLE
Labeling dims with units

### DISCUSSION THREAD

**Discussion by ercius** [pydata/xarray#6594] [Q&A] [answered]
Is there a way to label dimensions with units? For example you can create an image with an X and Y axis using a 2D ndarray:
```
import xarray as xr
import numpy as np
dd = xr.DataArray(np.zeros((10,10)), dims = ("X", "Y"), coords = {"X": np.linspace(0,1,10), "Y": np.linspace(0,1,10)})
```

However, its not clear in this data structure what the field of view in real space is. The X and Y axes could be meters or they could be Angstroms. Is there a way to indicate their units?

You can use `dd.attrs['units'] = "intensity"` but that labels the values for each pixel which will be shown on the colormap when you use `dd.plot()`. The X and Y labels should have a similar [meters] or [Angstroms] in the plot.

I tried to use 'pint' but that is not working (xarray '2022.3.0'). It seems like there should be a way since this is a fairly common and important aspect of a quantitative DataArray.

**Comment by andersy005** [comment, marked as answer]
> The X and Y labels should have a similar [meters] or [Angstroms] in the plot. 

@ercius, have you tried the following???

```python
In [5]: dd.X.attrs['units'] = 'm'

In [6]: dd.Y.attrs['units'] = 'm'

In [8]: dd.Y
Out[8]: 
<xarray.DataArray 'Y' (Y: 10)>
array([0.      , 0.111111, 0.222222, 0.333333, 0.444444, 0.555556, 0.666667,
       0.777778, 0.888889, 1.      ])
Coordinates:
  * Y        (Y) float64 0.0 0.1111 0.2222 0.3333 ... 0.6667 0.7778 0.8889 1.0
Attributes:
    units:    m
```


![Screen Shot 2022-05-11 at 1 11 58 PM](https://user-images.githubusercontent.com/13301940/167927611-88cbf6b7-c3bd-49e1-af83-9e9a5e19d192.png)

**Reply by ercius** [reply]
This is exactly what I was looking for! Thanks so much.
